### PR TITLE
Import domestic and international service areas from xlsx

### PR DESCRIPTION
**Pull request by gidjin**
## Description

Update the parser to parse out and import the domestic and international service area data.

## Reviewer Notes

Verify I'm pulling in the proper data and storing it in the proper tables.

## Setup

* Execute the `-h` option to see what options are available to run. Below are some examples of running the excel sheet. 
* Download the [FAKE Pricing Data sheet](https://docs.google.com/spreadsheets/d/1M1IdJfttPlasa_8NDJMsNIYNXksKMaR0bJsSkn7HpaA/edit) to process it

```sh
go run cmd/parse_ratengine_data_ghc/*.go  -h
go run cmd/parse_ratengine_data_ghc/*.go -filename "Pricing Template (2019-09-13) - FAKE DATA.xlsx" 
go run cmd/parse_ratengine_data_ghc/*.go -filename "Pricing Template (2019-09-13) - FAKE DATA.xlsx"   -save  -display -xlsxSheets="4"
```

Currently, the tests are not run during the build due to their location. This is part of the work that needs to be done but for now you can run the tests on the command line as below.

```sh
go test -v $(go list ./... | grep cmd\\/parse_ratengine_data_ghc)
```

## Code Review Verification Steps

* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/tree/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/tree/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] Request review from a member of a different team.
* [ ] Have the Pivotal acceptance criteria been met for this change?

## References

* [#168129992](https://www.pivotaltracker.com/story/show/168129992) for this change
* This builds on PR #2771

### CODE REVIEW

File: cmd/parse_ratengine_data_ghc/main.go [hunk 1]
```diff
@@ -525,4 +593,77 @@ var parseDomesticServiceAreaPrices processXlsxSheet = func(params paramConfig, s
...
+			ServiceAreaNumber: getInt(getCell(row.Cells, serviceAreaNumberColumn)),
+			Zip3s:             splitZip3s(getCell(row.Cells, zip3sColumn)),
+		}
+		// All the rows are consecutive, if we get to a blank one we're done
+		if domServArea.BasePointCity == "" {
+			break
+		} else if csvWriter != nil {
+			csvWriter.write(domServArea.toSlice())
+		}
+		domServArea.saveToDatabase(db)
```
Comment (reggieriser) [inline cmd/parse_ratengine_data_ghc/main.go]: If this fails, it looks like we currently log the error and then exit via a `log.Fatalf`.  How will this affect the database transaction that's open?  I would have expected `saveToDatabase` to return an err that then rolls back up the chain and gets picked up by the transaction which triggers a rollback.

File: cmd/parse_ratengine_data_ghc/shared.go [hunk 2]
```diff
@@ -257,3 +262,94 @@ func (dSA *domesticServiceAreaPrice) toSlice() []string {
...
+
+	values = append(values, dsa.BasePointCity)
+	values = append(values, dsa.State)
+	values = append(values, strconv.Itoa(dsa.ServiceAreaNumber))
+	values = append(values, strings.Join(dsa.Zip3s, ","))
+
+	return values
+}
+
+func (dsa *domesticServiceArea) saveToDatabase(db *pop.Connection) {
```
Comment (reggieriser) [inline cmd/parse_ratengine_data_ghc/shared.go]: I wonder if we should consider pushing these database queries into a service object and get it out of the `cmd` folder.

**Comment by jacquelineIO** [comment]
Need to update the description in this PR to show how to build and run with the new bin
```
make bin/ghc-pricing-parser
ghc-pricing-parser -h
ghc-pricing-parser --filename "Pricing Template (2019-09-13) - FAKE DATA.xlsx" --save
``` 
etc

Also I think we should add a flag to not save to the database. The `save` flag should probably be more descriptive to indicate `CSV` with a similar deal for the save the db option.

**Comment by jacquelineIO** [comment]
We've changed the scope of this story to cover reading the data and added a new story for writing the importer service object.

One thing, I think we should consider is letting the reading parser do some simple temp table saves "as-is" and letting the importer read from the temp tables as opposed to having to repeatedly re-read the xlsx sheets. The sheet parsing is slow. This also gives us separation from having the parser needing to know specifics about what the design is. The parser will parse into a temp table similar to how it is saving to CSV. 

Then the importer can read those tables and do the proper mapping it needs for clean inserts/updates into the real database tables

This will also us to create temp tables and run debugging and queries on the tables if we needed or wanted to do this separately from importing into the real database.

**Pull request merged**